### PR TITLE
V2 Settings Shell - Add Teams Section & UI fixes

### LINKED ISSUE #4105: /settings Sidebar needs formatting & links fixing (Compare with Figma)
[Compare with Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=17937%3A171647)

Image

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR adds teams pages to the settings shell. This PR also contains UI fixes.

Routing to the specific team pages will be added once those pages are completed

image

Fixes #4105

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- Sign in using the teampro user
- Go to the settings page
- The Seeded Team should appear on the side and all 5 team pages should be present
- Sign into the teampro2 user
- Go to the settings page
- Only the profile and members pages should be available

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/ui/v2/core/layouts/SettingsLayout.tsx [hunk 1]
```diff
@@ -89,18 +93,160 @@ const useTabs = () => {
...
+            <>
+              <div>
+                <div
+                  className="group flex h-9 w-64 flex-row items-center rounded-md px-3 py-[10px] text-sm font-medium leading-none text-gray-600 hover:bg-gray-100  group-hover:text-gray-700 [&[aria-current='page']]:bg-gray-200 [&[aria-current='page']]:text-gray-900"
+                  key={tab.name}>
+                  <tab.icon className="mr-[12px] h-[16px] w-[16px] self-start stroke-[2px] md:mt-0" />
+                  <p>{t(tab.name)}</p>
+                </div>
+              </div>
+              <div className="mt-2">
```
Comment (joeauyeung) [inline packages/ui/v2/core/layouts/SettingsLayout.tsx:135]: For section headers they should not be links so can't use VerticalTabItem

File: packages/ui/v2/core/layouts/SettingsLayout.tsx [hunk 2]
```diff
@@ -89,18 +93,160 @@ const useTabs = () => {
...
+    useState<{ teamId: number | undefined; teamMenuOpen: boolean }[]>();
+
+  const { data: teams } = trpc.useQuery(["viewer.teams.list"]);
+
+  useEffect(() => {
+    if (teams) {
+      const teamStates = teams?.map((team) => ({ teamId: team.id, teamMenuOpen: false }));
+      setTeamMenuState(teamStates);
+    }
+  }, [teams]);
```
Comment (joeauyeung) [inline packages/ui/v2/core/layouts/SettingsLayout.tsx:108]: Use this state to determine if the team specific collapsable menu should be open or closed

File: packages/ui/v2/core/layouts/SettingsLayout.tsx [hunk 3]
```diff
@@ -89,18 +93,160 @@ const useTabs = () => {
...
+                      <CollapsibleTrigger>
+                        <div
+                          className="flex h-9 w-64 flex-row items-center rounded-md px-3 py-[10px] text-sm font-medium leading-none hover:bg-gray-100  group-hover:text-gray-700 [&[aria-current='page']]:bg-gray-200 [&[aria-current='page']]:text-gray-900"
+                          onClick={() =>
+                            setTeamMenuState([
+                              ...teamMenuState,
+                              (teamMenuState[index] = {
+                                ...teamMenuState[index],
+                                teamMenuOpen: !teamMenuState[index].teamMenuOpen,
+                              }),
```
Comment (joeauyeung) [inline packages/ui/v2/core/layouts/SettingsLayout.tsx:180]: Update the specific team menu based on the team's index in the initial team query

File: packages/ui/v2/core/navigation/tabs/VerticalTabItem.tsx [hunk 4]
```diff
@@ -19,6 +19,7 @@ export type VerticalTabItemProps = {
   className?: string;
   isChild?: boolean;
   hidden?: boolean;
+  disableChevron?: boolean;
```
Comment (joeauyeung) [inline packages/ui/v2/core/navigation/tabs/VerticalTabItem.tsx:22]: Settings options should not show the chevron

**Review by zomars — APPROVED** [review]
Good enough for now but this can be further improved and abstracted.

Let's figure a way to incorporate this into VerticalTabs for 2.1

**Pull request merged**